apps/spark/build.gradle

```diff
@@ -1,38 +1,49 @@
 plugins {
-    id 'openhouse.java-conventions'
-    id 'openhouse.hadoop-conventions'
-    id 'openhouse.iceberg-conventions-1.2'
-    id 'openhouse.maven-publish'
+    // dependency in apps-spark-conventions
     id 'com.github.johnrengelman.shadow' version '7.1.2'
-}
-
-configurations {
-    // Excluding these libraries avoids competing implementations for LoggerFactory
-    // Standardizing on slf4j + log4j2 as implementation.
-    all*.exclude module : 'spring-boot-starter-logging'
-    all*.exclude module : 'logback-classic'
-    shadow.extendsFrom implementation
+    id 'openhouse.apps-spark-conventions'
 }
 
 ext {
+    log4jVersion = "2.18.0"
+
     sparkVersion = '3.1.1'
     icebergVersion = '1.2.0'
+    sparkVersionSuffix = "3.1"
+    openhouseSparkRuntimeModule = ":integrations:spark:spark-${sparkVersionSuffix}:openhouse-spark-runtime_2.12"
+    icebergSparkRuntimeModule = "org.apache.iceberg:iceberg-spark-runtime-3.1_2.12:${icebergVersion}"
+    tablesTestFixturesModule = ":tables-test-fixtures:tables-test-fixtures_2.12"
 }
 
 dependencies {
-    implementation project(':iceberg:openhouse:internalcatalog')
-    implementation project(':client:hts')
-    implementation project(':client:jobsclient')
-    implementation project(':client:tableclient')
-    implementation project(':client:secureclient')
-    implementation project(':services:common')
-    implementation project(':cluster:storage')
-    compileOnly (project(path: ':integrations:spark:spark-3.1:openhouse-spark-runtime_2.12')) {
+    compileOnly (project(path: openhouseSparkRuntimeModule)) {
         exclude group: 'io.netty'
         exclude group: 'org.apache.hadoop', module: 'hadoop-common'
         exclude group: 'org.apache.hadoop', module: 'hadoop-client'
     }
+
+    implementation "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"
     implementation project(':libs:datalayout')
+    implementation("org.apache.iceberg:iceberg-bundled-guava") {
+        version {
+            strictly("${icebergVersion}")
+        }
```

Comment:
are these really helpful ?  In practice I never find `force` or `strictly` covering everything that I want, since there could always be fatJar in the dependency Tree and these keywords don't work for them.

Reply:
strictly here overrides any dependencies of compaction that are defining a higher version of 1.5.2, such as 

services
cluster:storage
internalcatalog
htscatalog

an alternative to forcing the version to be LOWER than what is defined in transitive dependencies, is excluding all iceberg stuff from all of the dependencies above. but i think this is less readable and from what I understand, utilizing the resolution rules is preferred to exclusions

Reply:
changed to 2 from list instead